### PR TITLE
Nb/documentation and autocasting pixels

### PR DESCRIPTION
**Pull request by Nadav-Barak**
Small change to actual code - casting to 'uint8', if the type of pixels entered is not convertible to uint8 it is already caught in the validation method so no additional changes required.

Medium change to Object Detection doc, see in changes.

### CODE REVIEW

File: docs/source/user-guide/vision/data-classes/DetectionData.rst [hunk 1]
```diff
@@ -4,11 +4,9 @@
 The Object Detection Data Class
 ===============================
 
-The DetectionData is a :doc:`data class </user-guide/vision/data-classes/index>` represents a CV object detection task in deepchecks.
-It is a subclass of the :class:`~deepchecks.vision.VisionData` class and is used to load and preprocess data for an
-object detection task.
-The DetectionData class contains additional data and general methods intended for easy access to relevant metadata
-for object detection ML models validation.
+The DetectionData is a :doc:`data class </user-guide/vision/data-classes/index>` designated for object detection tasks.
```
Comment (noamzbr) [inline docs/source/user-guide/vision/data-classes/DetectionData.rst:7]:
```suggestion
The DetectionData is a :doc:`data class </user-guide/vision/data-classes/index>` designed for object detection tasks.
```

File: docs/source/user-guide/vision/data-classes/DetectionData.rst [hunk 2]
```diff
@@ -4,11 +4,9 @@
...
+It is a subclass of the :class:`~deepchecks.vision.VisionData` class and is used to load and preprocess data for object
```
Comment (noamzbr) [inline docs/source/user-guide/vision/data-classes/DetectionData.rst:8]:
```suggestion
It is a subclass of the :class:`~deepchecks.vision.VisionData` class and is used to help deepchecks load and interact with object detection data using a well defined format.
```

File: docs/source/user-guide/vision/data-classes/DetectionData.rst [hunk 3]
```diff
@@ -22,31 +20,34 @@ Accepted Label Format
 ---------------------
 Deepchecks' checks use the :func:`~deepchecks.vision.DetectionData.batch_to_labels` function in order to get the labels in the correct format.
 The accepted label format for is a a list of length N containing tensors of shape (B, 5), where N is the number
-of samples, B is the number of bounding boxes in the sample and each bounding box is represented by 5 values:
-``(class_id, x, y, w, h)``.
+of samples within a batch, B is the number of bounding boxes in the sample and each bounding box is represented by 5 values:
+``(class_id, x_min, y_min, w, h)``.
 
-    x and y are the coordinates (in pixels) of the upper left corner of the bounding box, w
+    x_min and y_min are the coordinates (in pixels) of the **lower left corner** of the bounding box, w
```
Comment (noamzbr) [inline docs/source/user-guide/vision/data-classes/DetectionData.rst:26]:
```suggestion
    x_min and y_min are the coordinates (in pixels) of the **top left corner** of the bounding box, w
```
I'm pretty sure it's top left...

File: docs/source/user-guide/vision/data-classes/DetectionData.rst [hunk 4]
```diff
@@ -22,31 +20,34 @@ Accepted Label Format
...
 Deepchecks' checks use the :func:`~deepchecks.vision.DetectionData.infer_on_batch` function in order to get the predictions of the model in the correct format.
 The accepted prediction format is a list of length N containing tensors of shape (B, 6), where N is the number
 of images, B is the number of bounding boxes detected in the sample and each bounding box is represented by 6
-values: ``[x, y, w, h, confidence, class_id]``.
+values: ``[x_min, y_min, w, h, confidence, class_id]``.
 
-    x and y are the coordinates (in pixels) of the upper left corner
-    of the bounding box, w and h are the width and height of the bounding box (in pixels), confidence is the
-    confidence of the model and class_id is the class id.
+    x_min,y_min,w and h represent the bounding box location as before, confidence is the confidence score given by the model to
```
Comment (noamzbr) [inline docs/source/user-guide/vision/data-classes/DetectionData.rst:39]: Not sure what I think, but consider writing explicitly and not "as before" - what if someone just wants to look at this and not at the label section?

File: docs/source/user-guide/vision/data-classes/DetectionData.rst [hunk 5]
```diff
@@ -86,48 +87,50 @@ Examples
...
+                A list of size N containing tensors of shape (B,5).
             """
 
             # each bbox in the labels is (class_id, x, y, x, y). convert to (class_id, x, y, w, h)
-            return [torch.stack(
-                   [torch.cat((bbox[0], bbox[1:3], bbox[4:] - bbox[1:3]), dim=0)
-                       for bbox in image])
-                    for image in batch[1]]
+            bboxes = []
+            for image in batch[1]:
```
Comment (noamzbr) [inline docs/source/user-guide/vision/data-classes/DetectionData.rst:105]: It's not an image it's label

File: docs/source/user-guide/vision/data-classes/DetectionData.rst [hunk 6]
```diff
@@ -86,48 +87,50 @@ Examples
...
 
         def batch_to_labels(self, batch):
-            """Convert a batch of labels to a tensor.
+            """Convert a batch bounding boxes to the required format.
 
             Parameters
             ----------
-            batch : torch.Tensor
-                The batch of labels to convert.
+            batch : torch.Tensor (N, B, 5)
```
Comment (noamzbr) [inline docs/source/user-guide/vision/data-classes/DetectionData.rst:94]: Technically in this example the back contains both image and label.
Also, is the original batch label a tensor? Given that B is different for each image?

File: docs/source/user-guide/vision/data-classes/DetectionData.rst [hunk 7]
```diff
@@ -86,48 +87,50 @@ Examples
...
 
         def infer_on_batch(self, batch, model, device):
             """Get the predictions of the model on a batch of images.
 
             Parameters
             ----------
-            batch : torch.Tensor
-                The batch of data.
+            batch : tuple
+                The batch of data, containing images and bounding boxes.
```
Comment (noamzbr) [inline docs/source/user-guide/vision/data-classes/DetectionData.rst:117]: can copy this line for previous function

**Review by noamzbr — COMMENTED** [review]
Important work! Added some comments about the changes in docs. @matanper Can I ask your opinion on the changes in the first two files?

File: docs/source/user-guide/vision/data-classes/DetectionData.rst (hunk 3 above)
Comment (ItayGabbay) [reply docs/source/user-guide/vision/data-classes/DetectionData.rst:26]: It's top left

File: docs/source/user-guide/vision/data-classes/DetectionData.rst (hunk 4 above)
Comment (Nadav-Barak) [reply docs/source/user-guide/vision/data-classes/DetectionData.rst:39]: Think it is the same as in image format that refers to a different page for details.

As above will be better?

Comment (noamzbr) [reply docs/source/user-guide/vision/data-classes/DetectionData.rst:39]: Yeah better IMO

**Review by noamzbr — APPROVED** [review]

**Review by ItayGabbay — APPROVED** [review]
